packages-web/maps/src/components/GoogleMap.tsx

```diff
@@ -0,0 +1,194 @@
+        }
+        addMarkers();
+    };
+
+    const addMarkers = (): void => {
+        markers.current.forEach(marker => marker.setMap(null));
+        markers.current = [];
+        if (props.locations && props.locations.length > 0) {
+            bounds = new google.maps.LatLngBounds();
+            markers.current = props.locations.reduce<google.maps.Marker[]>((markerArray, currentLocation) => {
+                const marker = addMarker(currentLocation);
+                if (marker) {
+                    markerArray.push(marker);
+                }
+                return markerArray;
+            }, []);
+        }
+        addCurrentLocation();
+        updateCamera();
+    };
+
+    const addCurrentLocation = (): void => {
+        if (props.currentLocation) {
+            const currentLocationMarker = addMarker(props.currentLocation);
+            if (currentLocationMarker) {
+                markers.current.push(currentLocationMarker);
+            }
+        }
+    };
+
+    const addMarker = (marker: Marker): google.maps.Marker | undefined => {
+        if (map.current) {
+            if (!bounds) {
+                bounds = bounds = new google.maps.LatLngBounds();
+            }
+            bounds.extend({
+                lat: marker.latitude,
+                lng: marker.longitude
+            });
+            const mapMarker = new google.maps.Marker({
+                position: {
+                    lat: marker.latitude,
+                    lng: marker.longitude
+                },
+                icon: marker.url
+                    ? {
+                          url: marker.url,
+                          scaledSize: new google.maps.Size(32, 32)
+                      }
+                    : "",
+                title: marker.title
+            });
+            if (marker.title) {
+                const infoContent = document.createElement("span");
+                infoContent.innerHTML = marker.title || "";
+                if (marker.onClick) {
+                    infoContent.style.cursor = "pointer";
+                    infoContent.onclick = marker.onClick;
+                }
+                const infoWindow = new google.maps.InfoWindow({
+                    content: infoContent
+                });
+                mapMarker.addListener("click", () => {
+                    infoWindow.open(map.current, mapMarker);
+                });
+            } else {
+                if (marker.onClick) {
+                    mapMarker.addListener("click", () => {
+                        marker.onClick!();
+                    });
+                } else {
+                    mapMarker.setClickable(false);
+                }
+            }
+            mapMarker.setMap(map.current);
+            return mapMarker;
+        }
+        return undefined;
+    };
+
+    const updateCamera = (): void => {
+        const { zoomLevel, autoZoom } = props;
+        setTimeout(() => {
```

Comment:
Is this because of race condition ?

Reply:
Yeap, removed because its not necessary anymore